daml/ci/build-unix.yml

```
@@ -8,10 +8,11 @@ parameters:
   scala_version: ''
 
 steps:
-  - bash: ci/dev-env-install.sh
+  - bash: daml/ci/dev-env-install.sh
```

```suggestion
  - bash: ci/dev-env-install.sh
    workingDirectory: daml
```

I think this makes more sense for inline scripts but github's ```` ```suggestion ```` doesn't work on those. In any case I'd also use it for "path" scripts for consistency.

With the suggestion from my other comment, this would be

```suggestion
  - bash: ci/dev-env-install.sh
    workingDirectory: ${{ variables.DAML_ROOT }}
```

---

Thanks for your comments!

Doing the change on this side has a couple benefits:

1. It allows us to "just add" the entire repo, with its (unchanged) history, as a merge commit in the canton repo.
2. It allows us to already test all of the CI configuration in a subfolder here, as a separate step from the actual merge, which reduces the risk of the move by not having to contend with all the issues at once.
3. Potentially, even if we didn't do the merge (which is still somewhat uncertain, hence my holding off on merging this), having the Bazel build isolated to a subfolder would allow us to start moving things out, should we want to.

For the `workingDirectory` option, I'm not sure I see the upside. The number of changes to make seems to be the same, with the key differences being that:

1. We use a platform-specific extension instead of a portable, well-known Bash command.
2. A piece of information absolutely crucial to understand how the script works would be moved to _after_ the script itself, even though for things to work it would have to be executed _before_.
3. Should we want to have scripts that work both in and out of the daml folder (which definitely will happen if we merge the repos; we'll have to do some cross-project stuff at some point), those will not be able to use the `workingDirectory` convention, which will not be immediately obvious as it's still going to appear (or not appear) at the end of the script.

Am I missing anything? Always happy to reconsider my blind spots.

Finally, for the variable, I don't think it would be all that useful as we have seven different entrypoints (seven different pipelines at the Azure level), so we'd have to define that variable in all of them. That can be done with an include, so it does indeed simplify things should we want to rename the folder down the line, but I don't think that's likely to happen often enough to warrant the indirection (which we pay for every time someone reads the code, vs. a small cost at the one point where we'd change the folder name again).

---

Thank you for your detailed response, your points about keeping the unchanged git history and testing CI before the actual merge convinced me. 

Regarding `workingDirectory` and the `DAML_ROOT` variable, my suggestion came from originally misunderstanding Azure's `steps.bash`: I assumed the scripts always ran in the same directory as the `.yml` file, then when I saw this PR prefixed {scripts with `cd daml`,script files with `daml/`}, I wanted to "restore" the relative-to-the-`.yml`-file behavior that was never there. But I guess that's really between Microsoft and me :)

```yml
# daml/ci/build-unix.yml
# ...
steps:
  - bash: daml/ci/dev-env-install.sh
#         ^^^^ relative to the 
    displayName: 'Build/Install the Developer Environment'
# ...
```

> GitHub is a bit too sensitive here and will consider any change to any moved file a conflict. I wouldn't pay too much attention to it; these are conflicts git can resolve automatically.

Ah, this one's on me for taking GitHub's ⚠️ at face value.

---

(Confusingly, the `template` "step" that lets you include another yml file is relative to the current yml file.)